.github/workflows/release.yaml

```diff
@@ -0,0 +1,177 @@
+                console.log(command, "exited with code", code)
+                if (code != 0) {
+                  core.setFailed(`Command ${command_str} exited with code ${code}`)
+                }
+              })
+            }
+
+            async function create_or_update_comment(identifier, comment_body) {
+              const comment_header = `<!-- ${identifier} -->\n`
+              const comment = comment_header + comment_body
+
+              const issue_number = ${{ steps.find_pr.outputs.pr }}
+              const owner = context.repo.owner
+              const repo = context.repo.repo
+
+              const existing_comments = (await github.rest.issues.listComments({
+                  issue_number, owner, repo
+              })).data
+
+              const comment_id = existing_comments.find(e => e.body?.startsWith(comment_header))?.id
+              const comment_obj = { owner, repo, body: comment }
+
+              if (comment_id) {
+                github.rest.issues.updateComment({ comment_id, ...comment_obj })
+              } else {
+                github.rest.issues.createComment({ issue_number, ...comment_obj })
+              }
+            }
+
+            async function create_github_release(crate_name, version) {
+              console.log("Creating a github release for", crate_name, version)
+
+              const { owner, repo } = context.repo
+              const tag_name = `${crate_name}-${version}`
+              const name = tag_name
+              const ref = `refs/tags/${tag_name}`
+              const sha = context.sha
+              const body = ""
+
+              await github.rest.git.createRef({ owner, repo, ref, sha })
+              await github.rest.repos.createRelease({ owner, repo, tag_name, name, body })
+            }
+
+            const is_dry_run = !is_merge_event
+            console.log("Dry run status is:", is_dry_run)
+
+            const crate_definitions = await globby("*/Cargo.toml")
+
+            const crates_to_publish = []
+            for (crate_definition of crate_definitions) {
+              console.log(`Processing ${crate_definition}`)
+
+              const data = toml.parse(fs.readFileSync(crate_definition, 'utf8'))
+              const name = data.package.name
+              const version_on_file = data.package.version
+              const version_on_crate_io = (await get_crate_details(name)).crate.max_version
+
+              if (version_on_file != version_on_crate_io) {
+                console.log("Crate", name, "need publishing")
+                crates_to_publish.push({ name, version_on_file, version_on_crate_io })
+
+                run_command_stream(["earthly", "--ci", "--org", "expressvpn", "--satellite", "wolfssl-rs", "--secret", "CARGO_REGISTRY_TOKEN", "+publish", `--PACKAGE=${name}`, `--DRY_RUN=${is_dry_run}`])
+                if (!is_dry_run) {
+                  await create_github_release(name, version_on_file)
+                }
```

Comment:
This creates a release for each crate, so if multiple crates are included in the ` for (crate_definition of crate_definitions) {` above, what happens?  i.e. what will https://github.com/expressvpn/wolfssl-rs/releases look like, will it be confusing wrt the different crates with different versions interleaved? If we release package A ten times after the last package B release will package  B's last release disappear off the end?

Do we actually want/need a GH release here, or would just a tag do?

Reply:
Package B's last release will disappear off the end, but it is fine as it will still be searchable. 

For your question "would just a tag do?", if the purpose is to retrieve the commit based on version number, then yes, a tag would do. But GitHub release seems to do more than just that:

1. Show up on people's feed when people "follow" a repository:

<img width="745" alt="image" src="https://github.com/expressvpn/wolfssl-rs/assets/106740022/208e74c3-71b3-44c3-8ab4-68b0d4c1688e">

2. Allow people to easily compare between releases

https://docs.github.com/en/repositories/releasing-projects-on-github/comparing-releases